File: 3-longest-substring-without-repeating-characters/3-longest-substring-without-repeating-characters.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
        int i =0;
        int j =0;
        int max = 0;
        HashSet<Character> Hs = new HashSet();
        while(j <s.length()){
            if(!Hs.contains(s.charAt(j))){
                Hs.add(s.charAt(j));
                j++;
                max = Math.max(Hs.size(), max); 
            }else{
                Hs.remove(s.charAt(i));
                i++;
            }
        }
    return max;
    
    
    }
}

File: 3-longest-substring-without-repeating-characters/3-longest-substring-without-repeating-characters.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        int i=0,j=0,ans=0;
        unordered_set<char>st;
        while(j<s.length() and i<s.length())
        {
           if(st.find(s[j])==st.end())
           {
               st.insert(s[j]);
               ans=max(ans,j-i+1);
               j++;
           }
        else
        {
            st.erase(s[i]);
            i++;
        }
        }
        return ans;
   
    }
};